### PR TITLE
Make all logging HikariCPConnectionManager SLS

### PR DESCRIPTION
**Pull request by Jolyon-S**
**Goals (and why)**:
* Not all arguments logged are correctly captured by (Un)safe args.

**Implementation Description (bullets)**:
* All arguments passed into logging messages (with the exception of exceptions) are captured in Safe or Unsafe args.

**Testing (What was existing testing like?  What have you done to improve it?)**:
N/A

**Concerns (what feedback would you like?)**:
Please verify that this fully adheres to internal specifications.

**Where should we start reviewing?**:
`HikariCPConnectionManager`.

**Priority (whenever / two weeks / yesterday)**:
This week.

### CODE REVIEW

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java [hunk 1]
```diff
@@ -237,7 +237,7 @@ public synchronized void close() {
                     if (log.isDebugEnabled()) {
                         log.debug(
                                 "Closing connection pool: {}",
-                                connConfig,
+                                UnsafeArg.of("connConfig", connConfig),
```
Comment (jkozlowski) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:240]: I think these all become unsafe anyway, but it's good hygiene.

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java [hunk 2]
```diff
@@ -237,7 +237,7 @@ public synchronized void close() {
                     if (log.isDebugEnabled()) {
                         log.debug(
                                 "Closing connection pool: {}",
```
Comment (jkozlowski) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:239]: Remove the '{}'

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java [hunk 3]
```diff
@@ -282,7 +282,8 @@ private State normalState() throws SQLException {
     private HikariDataSource getDataSourcePool() {
         // Print a stack trace whenever we initialize a pool
         if (log.isDebugEnabled()) {
-            log.debug("Initializing connection pool: {}", connConfig,
+            log.debug("Initializing connection pool: {}",
```
Comment (jkozlowski) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:285]: Ditto

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java [hunk 4]
```diff
@@ -304,7 +305,9 @@ private HikariDataSource getDataSourcePool() {
             }
         } catch (PoolInitializationException e) {
             log.error("[{}] Failed to initialize hikari data source: {}",
-                    connConfig.getConnectionPoolName(), connConfig.getUrl(), e);
+                    SafeArg.of("connectionPoolName", connConfig.getConnectionPoolName()),
```
Comment (jkozlowski) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:308]: Check if this has hostnames, cause I don't think these are allowed?

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java [hunk 5]
```diff
@@ -304,7 +305,9 @@ private HikariDataSource getDataSourcePool() {
             }
         } catch (PoolInitializationException e) {
             log.error("[{}] Failed to initialize hikari data source: {}",
```
Comment (jkozlowski) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:307]: Remove placeholders

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java [hunk 6]
```diff
@@ -345,7 +348,9 @@ private HikariPoolMXBean initPoolMbeans() {
         try {
             poolName = new ObjectName("com.zaxxer.hikari:type=Pool (" + connConfig.getConnectionPoolName() + ")");
         } catch (MalformedObjectNameException e) {
-            log.error("Unable to setup mBean monitoring for pool {}.", connConfig.getConnectionPoolName(), e);
+            log.error("Unable to setup mBean monitoring for pool {}.",
+                    SafeArg.of("connectionPoolName", connConfig.getConnectionPoolName()),
```
Comment (jkozlowski) [inline atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:352]: Same here, check conection pool what it contains

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java (hunk 4 above)
Comment (jkozlowski) [reply atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:308]: Yeah, so spec says: "Hostnames of Skylab-managed services (as they are already included in the Apollo envelope)" are safe. But I'm not convinced that e.g. if this is running with Oracle (would this be running on Oracle?) that we would be managing the hosts.

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java (hunk 2 above)
Comment (Jolyon-S) [reply atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:239]: Done

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java (hunk 3 above)
Comment (Jolyon-S) [reply atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:285]: Done

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java (hunk 5 above)
Comment (Jolyon-S) [reply atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:307]: Done

File: atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java (hunk 4 above)
Comment (Jolyon-S) [reply atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:308]: In the previous PR to add some safe logging, this was added as Safe in other places, so I was following suit. 

`getConnectionPoolName` is constructed as `getConnectionPoolIdentifier()-getConnId()`, where the former is `db-pool` and the latter is `atlas` as defaults. Obviously these can be overridden in the config, but I'd assume those two fields, given that they are just config (and not passwords or anything of that calibre either) would be fine, but maybe not.

Comment (jkozlowski) [reply atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java:308]: ok

**Pull request merged**